Applications/WebPortal/WebApp/tests/profile_page_tester/source/index.tsx

```diff
@@ -1,178 +1,130 @@
-            <WebPortal.TextField
-              displaySize={this.props.displaySize}
-              value={this.state.lastNameValue}
-              onInput={this.onTextInput}/>
-          </WebPortal.FormEntry>
-          <Dali.Padding size='30px'/>
-          <WebPortal.RolesField roles={this.state.someRoles}
-            onClick={this.onRoleClick}/>
-          <Dali.Padding size='30px'/>
-          <WebPortal.FormEntry name='Country' orientation={orientation} >
-            <WebPortal.CountrySelectionBox
-              displaySize={this.props.displaySize}
-              value={this.state.country}
-              onChange={this.changeCountry}
-              countryDatabase={this.state.countryDatabase}/>
-          </WebPortal.FormEntry>
-          <Dali.Padding size='30px'/>
-          <WebPortal.FormEntry name='Country' orientation={orientation} >
-            <WebPortal.CountrySelectionBox
-              readonly
-              displaySize={this.props.displaySize}
-              value={Nexus.DefaultCountries.AU}
-              onChange={this.changeCountry}
-              countryDatabase={this.state.countryDatabase}/>
-          </WebPortal.FormEntry>
-        </Dali.VBoxLayout>
-        <Dali.Padding size='18px'/>
+      <Dali.VBoxLayout width='100%' height='100%'>
+        <WebPortal.ProfilePage
+          roles={this.state.someRoles}
+          identity={this.state.identity}
+          displaySize={this.props.displaySize}
+          isSubmitEnabled={true}
+          submitStatus={this.state.statusMessage}
+          submitPasswordStatus={this.state.passwordStatusMessage}
+          hasError={this.state.hasError}
+          hasPasswordError={this.state.hasError}
+          account={this.state.account}
+          group={this.state.group}
+          countryDatabase={Nexus.buildDefaultCountryDatabase()}
+          />
         <div style={TestApp.STYLE.testingComponents}>
           <button tabIndex={-1}
-              onClick={this.toggleReadOnly}>
-            TOGGLE PHOTOFIELD READONLY
+            onClick={this.setStatusToNull}>
+            NO STATUS MESSAGES
           </button>
           <button tabIndex={-1}
-              onClick={this.changeImage}>
-            CHANGE IMAGE
+            onClick={this.setStatusToSuccessful}>
+            STATUS MESSAGES
+          </button>
+          <button tabIndex={-1}
+            onClick={this.setStatusToError}>
+            ERROR MESSAGES
           </button>
         </div>
-      </Dali.HBoxLayout>);
+      </Dali.VBoxLayout>);
   }
 
-  private onTextInput(value: string) {
-    this.setState({
-      lastNameValue: value
-    });
-  }
-
-  private onRoleClick(role: Nexus.AccountRoles.Role) {
-    if(this.state.someRoles.test(role)) {
-      this.state.someRoles.unset(role);
-    } else {
-      this.state.someRoles.set(role);
-    }
-    this.setState({someRoles: this.state.someRoles});
+  public componentDidMount() {
+    this.state.identity.photoId = TestApp.SOME_IMAGE;
```

Comment:
Construct a new ```AccountIdentity``` or copy the existing one, reassign all the values you need to, and then call ```setState```. Generally you should not mutate React state other than by calling ```setState```.